File: 1306. Jump Game III/Solution1306.py
from typing import List


class Solution1306: # 34-40
    def canReach(self, arr: List[int], start: int) -> bool:
        if start < 0 or start >= len(arr) or arr[start] < 0:
            return False
        if arr[start] == 0:
            return True
        arr[start] = -arr[start]
        return self.canReach(arr, start + arr[start]) or self.canReach(arr, start - arr[start])

File: 1306. Jump Game III/Solution1306.cpp
#include<iostream>
#include<string>
#include<vector>

using namespace std;

class Solution1306 {
public:
    bool canReach(vector<int>& arr, int start) {
        if (start < 0 || start >= arr.size() || arr[start] < 0) {
            return false;
        }
        if (arr[start] == 0) {
            return true;
        }
        arr[start] = -arr[start];
        return canReach(arr, start-arr[start]) || canReach(arr, start+arr[start]);
    }
};